lms/static/js/dashboard/legacy.js

```diff
@@ -105,19 +105,64 @@
          });
 
          $('.action-unenroll').click(function(event) {
+
              var element = $(event.target);
-             var track_info = element.data('track-info');
+
+             var can_unenroll =  $('#course-context-data').attr('data-course_can_unenroll') == 'True' ? true:false;
+             var is_paid_course = $('#course-context-data').attr('data-course_is_paid_course') == 'True'? true:false;
+             var cert_name_long = $('#course-context-data').attr('data-course_cert_name_long');
+             var enrollment_mode = $('#course-context-data').attr('data-course_enrollment_mode');
+
+             var course_id = element.data('course-id');
              var course_number = element.data('course-number');
              var course_name = element.data('course-name');
-             var cert_name_long = element.data('cert-name-long');
-             $('#track-info').html(interpolate(track_info, {
-                 course_number: "<span id='unenroll_course_number'>" + course_number + '</span>',
-                 course_name: "<span id='unenroll_course_name'>" + course_name + '</span>',
-                 cert_name_long: "<span id='unenroll_cert_name'>" + cert_name_long + '</span>'
-             }, true));
-             $('#refund-info').html(element.data('refund-info'));
-             $('#unenroll_course_id').val(element.data('course-id'));
-             edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+             var show_refund_option="";
+
+             var request = $.ajax({
+                      url: "/course_refund_status/?course_id="+ encodeURIComponent(course_id),
+                      method: "GET",
+                      dataType: "json"
+                    });
+
+                request.success(function( data ) {
+                    if (data.success){
+                        show_refund_option = data.course_refundable_status;
+
+                        var dialog_message_attr = set_dialog_attributes(can_unenroll, is_paid_course, cert_name_long,
+                                        course_number, course_name, enrollment_mode, show_refund_option);
+
+                        $('#track-info').html(interpolate(dialog_message_attr['data-track-info'], {
```

Comment:
Can we set the #track-info outside of the api request? Looks unrelated to what information we need to get from the xhr request.